File: src/main/java/org/lvtinger/paladin/account/api/AccountService.java
package org.lvtinger.paladin.account.api;

import org.lvtinger.paladin.value.Result;

public interface AccountService {
    Result<String> register(String username, String password);
    Result<String> login(String username, String password);
    Result<Boolean> changePassword(Long accountId, String password, String replacer);
}

File: src/main/kotlin/org/risesun/paladin/provider/api/AccountService.kt
package org.risesun.paladin.provider.api

import org.risesun.paladin.value.ResponseMessage

interface AccountService {
    fun exists(username: String): Boolean
    fun token(token: String): Long?
    fun login(username: String, password: String): ResponseMessage<String?>
    fun register(username: String, password: String): ResponseMessage<String?>
    fun change(id: Long, original: String, password: String):ResponseMessage<Boolean>
}